plugins/glustershd/init.go

```diff
@@ -18,24 +20,34 @@ func (p *Plugin) Name() string {
 func (p *Plugin) RestRoutes() route.Routes {
 	return route.Routes{
 		route.Route{
-			Name:        "GlustershEnable",
-			Method:      "POST",
-			Pattern:     "/volumes/{name}/heal/enable",
-			Version:     1,
-			HandlerFunc: glustershEnableHandler},
+			Name:         "SelfHealInfo",
+			Method:       "GET",
+			Pattern:      "/volumes/{name}/{opts}/heal-info",
+			Version:      1,
+			ResponseType: utils.GetTypeString((*glustershdapi.HealInfo)(nil)),
+			HandlerFunc:  selfhealInfoHandler},
 		route.Route{
-			Name:        "GlustershDisable",
-			Method:      "POST",
-			Pattern:     "/volumes/{name}/heal/disable",
-			Version:     1,
-			HandlerFunc: glustershDisableHandler},
+			Name:         "SelfHealInfo2",
+			Method:       "GET",
+			Pattern:      "/volumes/{name}/heal-info",
+			Version:      1,
+			ResponseType: utils.GetTypeString((*glustershdapi.HealInfo)(nil)),
+			HandlerFunc:  selfhealInfoHandler},
 	}
 }
 
 // RegisterStepFuncs registers transaction step functions with
 // Glusterd Transaction framework
 func (p *Plugin) RegisterStepFuncs() {
-	transaction.RegisterStepFunc(txnSelfHealStart, "selfheal-start")
-	transaction.RegisterStepFunc(txnSelfHealdUndo, "selfheald-undo")
-	transaction.RegisterStepFunc(txnSelfHealStop, "selfheal-stop")
+	var sfs = []struct {
+		sf   transaction.StepFunc
+		name string
+	}{
+		{txnHealInfo, "selfheal-info"},
+		{txnValidateVolNameType, "validate-vol-name-type"},
```

Comment:
Both validate steps can be merged into one validate step.

Also, prefix step names with `selfheal`

Reply:
Actually, validations can happen in handler after taking lock.